### ISSUE TITLE
Contacts Backup: Introduce Time Intervall for backups

### ISSUE THREAD

**Issue by MarkusH1975** [nextcloud/android#2296]
### Actual behaviour
- Daily backup of contacts is done right now

### Expected behaviour
- Possibility to configure time intervall for backups
   May be I want to have a backup only weekly or monthly
   Beacause normally I don't change a lot.

- Another good thing would be to automatically delete old backups
   May be older than 3 month or what else
 
Nextcloud app version:
V3.0.3 

Nextcloud server version:
V13

**Comment by tobiasKaminsky** [comment]
I am unsure if a job can run this few (once per week / month). If the app is terminated/phone restarted, the job will queued again and starting again waiting a week/month. Therefore I would keep it this way.

Internally there is already the possibility to expire.
So this will mainly need a UI.

![2018-03-08-144842](https://user-images.githubusercontent.com/5836855/37154329-ede6496e-22df-11e8-9df7-04caa14ec537.png)

I suggest to have a chooser with prefixed values "week, 1 month, 2 months, 3 months, 6 months, 12 months/year".
It should be beneath "last backup".

@nextcloud/designers what do you think?

**Comment by Seferi** [comment]
This is definitely a needed feature.. I have tons off contacts backup in the hidden folder and deleting then manually is a pain..

**Comment by xXSTrikeXx** [comment]
@Seferi A suggestion for deletion: Select all and then deselect the latest which you want to keep 😉

**Comment by MarkusH1975** [comment]
@tobiasKaminsky prefixed values sounds good.

Interval for Backups: 
"every day, every week, every 2 weeks, every month, every 2 month, every 3 month" 

I personally wound not go to more than 3 month. I think at some point it makes no sense any more.

@Seferi, @xXSTrikeXx , what about prefixed values for deletion:
Number of backups to keep:
"1, 2, 3, 5, 10, 20, 30, 50,100, all"

Since the backups have a time stamp like "2018-03-07_09-30-13.vcf" it should be possible to sort the filenames and count the number to keep and delete the others.

If a certain number of backups is kept, there is the advantage, not to run into a strange influence between time interval to do backups and a time interval to keep backups, this could be problematic.

**Comment by jancborchardt** [comment]
@tobiasKaminsky yes, should be a setting called »Backup interval« with: daily, weekly, monthly. (More doesn’t really make sense, as then it gets really arbitrary.)

It should be below the »Automatic backup« switch and also be hidden unless that is turned on cause it’s irrelevant otherwise.

The »Last backup« section isn’t really a setting, and is probably better as smaller and greyed out info text below the »Back up now« button. Where it would then say »Last backup: 2 days ago«

**Comment by CalebBurton** [comment]
I'd love to see this feature. I just realized my ~1MB contact backup files have filled up an entire gigabyte... I had assumed it was a single file that was replaced daily and I was shocked to find thousands of identical and nearly-identical files in there.

**Comment by stefan-niedermann** [comment]
> I am unsure if a job can run this few (once per week / month).

Just a few notes from me: `PeriodicWorkRequest` from the `AndroidX WorkManager` has explicitly set a hardcoded [`MIN_PERIODIC_INTERVAL_MILLIS`](https://android.googlesource.com/platform/frameworks/support/+/androidx-1.0-dev/work/workmanager/src/main/java/androidx/work/PeriodicWorkRequest.java#34). I am not able to find a `MAX` value, though, so guess this might work reliable.

@fam4r & @CalebBurton Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue. Please don't comment if you have no relevant information to add as this makes it hard to read and follow an issue (and thus will slow down the development instead of speeding it up.)
